docs/classic-ui/mockup.md

```diff
@@ -52,6 +52,80 @@ You can check the full list of available features using the `-l` parameter:
 plonecli -l
 ```
 
+## Create a custom pattern
+
+To create a custom {term}`pattern` in your add-on, use the `mockup_pattern` {term}`bobtemplate`:
+
+```shell
+cd project.addon
+plonecli add mockup_pattern
+```
+
+Next, enter your pattern name without the `pat-` prefix.
+
+```shell
+--> Pattern name (without “pat-” prefix) [my-pattern]: testpattern
+```
+
+This creates the necessary JavaScript resources and webpack configuration for you, as shown in the following file system tree diagram.
+
+```text
+...
+├── resources
+|   ├── pat-testpattern
+|   |   ├── documentation.md
+|   |   ├── testpattern.js
+|   |   ├── testpattern.scss
+|   |   ├── testpattern.test.js
+│   ├── bundle.js
+│   ├── index.html
+│   ├── index.js
+├── package.json
+├── webpack.config.js
+...
+```
+
+All your pattern JavaScript code goes into {file}`resources/pat-testpattern/testpattern.js`.
+SCSS files can be imported, too, since webpack provides the `sass-loader` module.
+
+Next, install the npm packages using {term}`yarn`.
+
+```shell
+yarn install
+```
+
+When you finish writing your JavaScript code, you have to build the bundle with the following command.
+
+```shell
+yarn build
+```
+
+This creates the webpack chunks and the JavaScript bundle files in your add-on package.
+
+```text
+...
+├── src
+|   ├── project
+|   |   ├── addon
+|   |   |   ├── browser
+|   |   |   |   ├── static
+|   |   |   |   |   ├── bundles
+|   |   |   |   |   |   ├── chunks
+|   |   |   |   |   |   ├── addon-remote.min.js
+|   |   |   |   |   |   ├── addon-remote.min.js.map
+|   |   |   |   |   |   ├── addon.min.js
+|   |   |   |   |   |   ├── addon.min.js.map
+```
+
+Note that `plonecli` also creates an XML file in {file}`src/project/addon/profiles/default/registry/bundles.xml`, which registers the {file}`addon-remote.min.js` file in the resource registry.
+
+```{important}
+You must re-import your profile with an upgrade step if you installed your add-on in Plone before adding the pattern.
+```
+
+You can test your pattern now with the browser view `@@addon-pattern-demo` (see {file}`/src/project/addon/browser/pattern-demo.pt`)
```

Comment:
Can you explain further? Do I load something in a web browser? What does the file have to do with testing?

Reply:
ah ok ... its not really a test but a demo setup how to integrate your patter in a template.
This is accessible through the browser with `http://localhost:8080/Plone/@@addon-pattern-demo` depending on the development port setup and the Plone Site ID.

Not sure if we should mention it at all but instead explain how its done with css class (see below)

Reply:
I'll go with what you think is best. We don't need to document all usage, but the most useful to developers. I added an example of the class, and if that is "best", then let's go with it.

Reply:
I updated the info for the demo browser view. see https://github.com/plone/documentation/pull/1748/commits/b25cf80328a3fbcc911d702a7d2195a4a2858456 ...  lets keep it that way.